### ISSUE TITLE
[A11Y] Keyboard hovering over the title of the website does not apply focus colours correctly

### ISSUE THREAD

**Issue by wardt7** [UKHomeOffice/engineering-guidance-and-standards#601]
**Describe the accessibility issue**
When you hover over the title of the site, the background focus colour is incorrectly applied as seen below:

<img width="1516" height="208" alt="Image" src="https://github.com/user-attachments/assets/852e7004-353f-4111-bf52-dcd9baffe75c" />

The correct result should be for the focus yellow to apply across the whole background, and for all content to be made GDS black:

<img width="1725" height="197" alt="Image" src="https://github.com/user-attachments/assets/a67ab7f1-4b65-4919-8be4-ad9ecf23ca1f" />

**To Reproduce**
- Focus on the title element, typically through keyboard navigation to the title

**Expected behavior**
The background of the entire link should become yellow, with all content becoming black.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser: Edge
 - Version: Latest

**Any assistance technologies used**
- Navigation method: Keyboard

**Additional context**

I found this bug whilst working on a fork of this project for use within Ofqual (https://github.com/[OfqualGovUK/ofqual-standards-patterns](https://github.com/OfqualGovUK/ofqual-standards-patterns)) and we have made a fix ourselves that resolves the problem (as seen at https://digital-framework.ofqual.gov.uk/). The main elements of the fix are:

- Setting the display of the header link class to inline-block. This forces the <a> tag to become the height of its children
- Explicitly setting the background-color and colors on the header link class as per the focus guidelines
- Using two different logo assets; one for use in the hover and one for normal circumstances. You can use some CSS then to control when these appear on focus. This _is_ different to how GOV.UK does it, but that's because they use a proper <svg> element instead that lets them manipulate the colours directly.
- Note that I did have to do a few further adjustments to remove positioning adjustments around the logo to get the background to appropriately fill; you may need to do the same depending upon your logo requirements.
- I did _not_ run the a11y cypress tests for our fix; we cannot run cypress on our work laptops and we're currently converting to using playwright instead, so you may want to take this fix with a pinch of salt

CSS Snippet:

```css
.govuk-header__link, .govuk-header__link:visited, .govuk-header__link:active, .govuk-header__link:hover {
  color: #fff;
  display: inline-block;
}

.govuk-header__link:focus {
  background-color: #ffdd00;
  color: #0b0c0c;
}

// This hide class and child sections get
// used to swap around the Ofqual logo for when
// the main link is hovered over

.hide {
  display: none
}

.govuk-header__link:focus img:first-child
{
  display: none;
}

.govuk-header__link:focus img:last-child
{
  display: inline-block
}
```

.eleventy.js snippet

```js
logotype: {
    html:
        '<span class="govuk-header__logotype">' +
        '  <img src="/assets/logos/OfqualLogoWhite.svg" height="64px" alt="Ofqual Logo">' +
        '  <img class="hide" src="/assets/logos/OfqualLogoBlack.svg" height="64px" alt="Ofqual Logo">' +
        '</span>'
},
```
